plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go

```diff
@@ -286,14 +282,7 @@ func buildControllerRoles() ([]rbacv1.ClusterRole, []rbacv1.ClusterRoleBinding)
 			rbacv1helpers.NewRule("get", "list", "watch", "update").Groups(legacyGroup).Resources("persistentvolumeclaims").RuleOrDie(),
 			rbacv1helpers.NewRule("update").Groups(legacyGroup).Resources("persistentvolumeclaims/status").RuleOrDie(),
 			rbacv1helpers.NewRule("list", "watch", "get", "create", "delete").Groups(legacyGroup).Resources("pods").RuleOrDie(),
-
-			// glusterfs
 			rbacv1helpers.NewRule("get", "list", "watch").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
```

Comment:
The PV controller requires storageclass permissions, so keep it.

Reply:
do you have a link to this? knowing how you audited what was still in use would help build confidence that the other things being removed from this role are not in use by the controller in some other way

Reply:
1. PV controller: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/pv_controller_base.go#L63
   - informers: pv, pvc, sc, pod, node (list, watch)
     - It is so strange that `persistent-volume-binder` clusterrole doesn't contain the watch verb for node resource. And the list verb is added in the openstack section. @jsafrane @msau42 @liggitt should I add the watch and list verb for node resource in the `persistent-volume-binder` clusterrole?
     - And the sc permission is added in the glusterfs section.
   - kubeClient: pv (get, update), pvc (get, update)
   - others: event
   - volume plugins: https://github.com/kubernetes/kubernetes/blob/beb696c2c9467dbc44cbaf35c5a4a3daf0321db3/cmd/kube-controller-manager/app/plugins.go#L82 used to provision/reclaim the volume.
     - hostpath: 
       - recycler: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/hostpath/host_path.go#L156 (kube client)
       - deleter: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/hostpath/host_path.go#L354 (no kube client used)
       - provisioner: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/hostpath/host_path.go#L293 (no kube client used)
     - nfs: 
       - recycler: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/nfs/nfs.go#L156 (kube client)
     - portworx: 
       - deleter: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/portworx/portworx.go#L383 (use kube client to get *service* when it delete the volume https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/portworx/portworx_util.go#L289)
       - provisioner: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/portworx/portworx.go#L394 (use kube client to get *service* when it create the volume https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/portworx/portworx_util.go#L289)
     These volume plugins doesn't access to `endpoints` and `nodes` resources. I can't find the code that it uses kube client to access the specific resource.

2. expand controller: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/expand/expand_controller.go#L100
   - informers: pvc (list, watch)
   - kubeClient: pv (get) https://github.com/kubernetes/kubernetes/blob/beb696c2c9467dbc44cbaf35c5a4a3daf0321db3/pkg/controller/volume/expand/expand_controller.go#L349
   - others: event
   - volume plugins: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/plugins.go#L68 used to expand the volume.
     - portworx: 
       - expander: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/portworx/portworx.go#L204 (use kube client to get *service* when it expand the volume https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/portworx/portworx_util.go#L289)
       -  And the get verb for service is added in the glusterfs section.
     - fc: it doesn't implement the expander interface.
     These volume plugins doesn't access to `endpoints`, `storageclasses` and `secrets` resources. I can't find the code that it uses kube client to access the specific resource.

Reply:
/hold

Reply:
@jsafrane @liggitt  can you review it again? I pushed some changes.

Reply:
I'd like storage folks to do the first pass review here since they have domain knowledge of the controller loops

Reply:
/hold cancel

Reply:
Thanks a lot.

/cc @liggitt 

Please review it again.